legacy_packages/sdk/src/evm/common/feature-detection/fetchPreDeployMetadata.ts

```diff
@@ -15,15 +17,61 @@ import { fetchRawPredeployMetadata } from "./fetchRawPredeployMetadata";
 export async function fetchPreDeployMetadata(
   publishMetadataUri: string,
   storage: ThirdwebStorage,
+  compilerOptions?: CompilerOptions,
 ): Promise<PreDeployMetadataFetched> {
   const rawMeta = await fetchRawPredeployMetadata(publishMetadataUri, storage);
-  const deployBytecode = await (
-    await storage.download(rawMeta.bytecodeUri)
-  ).text();
-  const parsedMeta = await fetchContractMetadata(rawMeta.metadataUri, storage);
+  let bytecodeUri;
+  let metadataUri;
+
+  // TODO: Add an invariant, throw if no compilers in rawMeta in case compilerOptions are provided.
+  // Keeping it this way until all contracts are republished with new metadata.
```

Comment:
what about 3rd party contracts? Will those also be republished before we add this here?

Reply:
For now applicable only to zksync deployments. Also, if not republished, it will fallback to old publish metadata uri etc.